examples/sparse/sign.py

```diff
@@ -0,0 +1,82 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.data import CoraGraphDataset
+from dgl.mock_sparse import create_from_coo, diag, identity
+from torch.optim import Adam
+
+
+def sign(A, X, num_hops):
+    X_sign = [X]
+    for _ in range(num_hops):
+        X = A @ X
+        X_sign.append(X)
+    return X_sign
+
+
+class Model(nn.Module):
+    def __init__(self, in_size, num_hops, num_classes, hidden_size=256):
+        super().__init__()
+
+        num_feats = num_hops + 1
+        self.linear = nn.ModuleList(
+            [nn.Linear(in_size, hidden_size) for _ in range(num_feats)]
+        )
+        self.pred = nn.Linear(hidden_size * num_feats, num_classes)
+
+    def forward(self, X):
+        X = torch.cat([self.linear[i](X[i]) for i in range(len(X))], dim=1)
```

Comment:
Nesting this together made it hard to read (for me).
Not sure we can unnest the code and add comment to guide readers to understand the code better.